src/nvme/fabrics.c

```diff
@@ -771,7 +771,10 @@ static int uuid_from_device_tree(char *system_uuid)
 	f = open(PATH_UUID_IBM, O_RDONLY);
 	if (f < 0)
 		return -ENXIO;
-	len = read(f, system_uuid, 512);
+
+	/* uuid is max 37 chars long,
+	 * e.g. 1b4e28ba-2fa1-11d2-883f-0016d3cca427 + \0 */
+	len = read(f, system_uuid, 37);
```

Comment:
We need to forcibly insert a terminating 'NULL' character after the read; we have no idea _what_ we read, so there is no guarantee that the string is properly terminated.